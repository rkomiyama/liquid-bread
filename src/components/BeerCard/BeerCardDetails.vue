<template>
  <v-layout>
    <v-flex xs6>
      <h3>Ingredients</h3>
      <h4>Malt</h4>
      <v-layout v-for="malt in ingredients.malt" :key="malt.name">
        <v-flex xs6>{{ malt.name }}</v-flex>
        <v-flex xs6>{{ malt.amount.value }} {{ malt.amount.unit }}</v-flex>
      </v-layout>
      <h4>Hops</h4>
      <v-layout v-for="hops in ingredients.hops" :key="hops.name + hops.add">
        <v-flex xs6>{{ hops.name }} ({{ hops.add }})</v-flex>
        <v-flex xs6>{{ hops.amount.value }} {{ hops.amount.unit }}, {{ hops.attribute }}</v-flex>
      </v-layout>
      <h4>Yeast</h4>
      <div>{{ ingredients.yeast }}</div>
    </v-flex>
    <v-flex xs6>
      <h3>Method</h3>
      <h4>Fermentation</h4>
      <v-layout>
        <v-flex xs6>Temp</v-flex>
        <v-flex xs6>{{ method.fermentation.temp.value }} {{ method.fermentation.temp.unit }}</v-flex>
      </v-layout>
      <h4>Mash temp</h4>
      <div v-for="(mash_temp, i) in method.mash_temp" :key="i">
        <v-layout>
          <v-flex xs6>Duration</v-flex>
          <v-flex xs6>{{ mash_temp.duration }}</v-flex>
        </v-layout>
        <v-layout>
          <v-flex xs6>Temp</v-flex>
          <v-flex xs6>{{ mash_temp.temp.value }} {{ mash_temp.temp.unit }}</v-flex>
        </v-layout>
      </div>
      <h4>Twist</h4>
      <div>{{ method.twist ? method.twist : 'N/A' }}</div>
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  name: "BeerCardDetails",
  props: {
    ingredients: {
      type: Object,
      default: () => {}
    },
    method: {
      type: Object,
      default: () => {}
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
