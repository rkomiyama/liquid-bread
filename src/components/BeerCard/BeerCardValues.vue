<template>
  <v-layout>
    <v-flex xs6>
      <h3>ABV</h3>
      <h3>IBU</h3>
      <h3>EBC</h3>
      <h3>SRM</h3>
      <div>First brewed</div>
      <div>Volume</div>
      <div>Attenuation level</div>
      <div>pH</div>
      <div>Boil volume</div>
      <div>Target OG</div>
      <div>Target FG</div>
      <div>Contributed by</div>
    </v-flex>
    <v-flex xs6>
      <h3>{{ beer.abv ? beer.abv : 'N/A' }}%</h3>
      <h3>{{ beer.ibu ? beer.ibu : 'N/A' }} BU</h3>
      <h3>{{ beer.ebc ? beer.ebc : 'N/A' }}</h3>
      <h3>{{ beer.srm ? beer.srm : 'N/A' }}</h3>
      <div>{{ beer.first_brewed ? beer.first_brewed : 'N/A' }}</div>
      <div>{{ beer.volume ? `${beer.volume.value} ${beer.volume.unit}` : 'N/A' }}</div>
      <div>{{ beer.attenuation_level ? beer.attenuation_level : 'N/A' }}</div>
      <div>{{ beer.ph ? beer.ph : 'N/A' }}</div>
      <div>{{ beer.boil_volume ? `${beer.boil_volume.value} ${beer.boil_volume.unit}` : 'N/A' }}</div>
      <div>{{ beer.target_og ? beer.target_og : 'N/A' }}</div>
      <div>{{ beer.target_fg ? beer.target_fg : 'N/A' }}</div>
      <div>{{ beer.contributed_by ? beer.contributed_by : 'N/A' }}</div>
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  name: "BeerCardValues",
  props: {
    beer: {
      type: Object,
      default: () => {}
    }
  }
};
</script>

<style lang="scss" scoped>
</style>
