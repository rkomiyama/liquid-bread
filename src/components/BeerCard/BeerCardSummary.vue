<template>
  <div>
    <v-card-title>
      <h1>
        {{ beer.name }}
        <span class="tagline font-italic">{{ beer.tagline }}</span>
      </h1>
    </v-card-title>
    <v-card-text class="text-xs-left pt-0">
      <v-layout>
        <v-flex xs6>
          <BeerCardValues :beer="beer"/>
        </v-flex>
        <v-flex xs6>
          <div class="font-weight-bold">Description</div>
          <div>{{ beer.description }}</div>
          <div class="font-weight-bold">Brewers tips</div>
          <div>{{ beer.brewers_tips }}</div>
          <div class="font-weight-bold">Food pairing</div>
          <div v-for="(food, i) in beer.food_pairing" :key="i">{{ food }}</div>
        </v-flex>
      </v-layout>
      <v-slide-y-transition>
        <div v-show="showMore">
          <v-divider></v-divider>
          <BeerCardDetails :ingredients="beer.ingredients" :method="beer.method"/>
        </div>
      </v-slide-y-transition>
      <v-flex text-xs-center>
        <v-btn flat @click="showMore = !showMore">{{ showMore ? 'Show Less' : 'Show More' }}</v-btn>
      </v-flex>
    </v-card-text>
  </div>
</template>

<script>
import BeerCardValues from "./BeerCardValues";
import BeerCardDetails from "./BeerCardDetails";

/**
 * Displays the description, brewers tips, and food pairing for the beer.
 */
export default {
  name: "BeerCardSummary",
  data() {
    return {
      showMore: false
    };
  },
  props: {
    /**
     * Beer data object for the beer
     */
    beer: {
      type: Object
    }
  },
  components: {
    BeerCardValues,
    BeerCardDetails
  }
};
</script>

<style lang="scss" scoped>
.tagline {
  color: grey;
}
</style>

<docs>
```jsx
const mockData = require('../../../tests/mockData/beerData.json');
<BeerCardSummary :beer="mockData" />
```
</docs>