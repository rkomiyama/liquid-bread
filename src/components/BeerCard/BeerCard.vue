<template>
  <v-card flat class="beer-card">
    <v-layout>
      <v-flex xs2 pa-2>
        <v-img :src="beer.image_url" class="beer-image"></v-img>
      </v-flex>
      <v-flex xs10>
        <BeerCardSummary :beer="beer"/>
      </v-flex>
    </v-layout>
  </v-card>
</template>

<script>
import BeerCardSummary from "./BeerCardSummary";

/**
 * Container component for beer data.
 */
export default {
  name: "BeerCard",
  props: {
    /**
     * Beer data object for the beer
     */
    beer: {
      type: Object
    }
  },
  components: {
    BeerCardSummary
  }
};
</script>

<style lang="scss" scoped>
.beer-card {
  background-color: #343334;
}
.beer-image {
  background-color: white;
}
</style>

<docs>
```jsx
const mockData = require('../../../tests/mockData/beerData.json');
<BeerCard :beer="mockData" />
```
</docs>