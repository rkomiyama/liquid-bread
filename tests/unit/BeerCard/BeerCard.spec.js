import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import BeerCard from "@/components/BeerCard/BeerCard.vue";
import beerData from "#/mockData/beerData.json";

Vue.use(Vuetify);

let wrapper;

describe("BeerCard.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(BeerCard, { propsData: { beer: beerData } });
  });

  it("BeerCard has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
