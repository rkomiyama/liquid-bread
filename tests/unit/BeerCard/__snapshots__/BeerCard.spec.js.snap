// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard.vue BeerCard has same HTML structure 1`] = `
<v-card-stub tag="div" flat="true" class="beer-card">
  <v-layout-stub tag="div">
    <v-flex-stub tag="div" xs2="" pa-2="">
      <v-img-stub src="https://images.punkapi.com/v2/keg.png" position="center center" transition="fade-transition" class="beer-image"></v-img-stub>
    </v-flex-stub>
    <v-flex-stub tag="div" xs10="">
      <BeerCardSummary-stub beer="[object Object]"></BeerCardSummary-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-card-stub>
`;
